src/trafficcontroller/internal/proxy/doppler_proxy.go

```diff
@@ -45,177 +36,51 @@ type grpcConnector interface {
-	p.serveAppLogs("recentlogs", mux.Vars(r)["appID"], w, r)
-	// metric-documentation-v1: (dopplerProxy.recentlogsLatency) USELESS metric which measures nothing of value
-	sendLatencyMetric("recentlogs", time.Now())
-}
-
-func (p *DopplerProxy) containermetrics(w http.ResponseWriter, r *http.Request) {
-	p.serveAppLogs("containermetrics", mux.Vars(r)["appID"], w, r)
-	// metric-documentation-v1: (dopplerProxy.containermetricsLatency) USELESS metric which measures nothing of value
-	sendLatencyMetric("containermetrics", time.Now())
-}
-
-func (p *DopplerProxy) setcookie(w http.ResponseWriter, r *http.Request) {
-	p.serveSetCookie(w, r, p.cookieDomain)
-}
-
-func (p *DopplerProxy) serveFirehose(firehoseSubscriptionId string, writer http.ResponseWriter, request *http.Request) {
-	authToken := getAuthToken(request)
-
-	authorized, err := p.adminAuthorize(authToken)
-	if !authorized {
-		writer.Header().Set("WWW-Authenticate", "Basic")
-		writer.WriteHeader(http.StatusUnauthorized)
-		fmt.Fprintf(writer, "You are not authorized. %s", err.Error())
-		return
-	}
-
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
+	r := mux.NewRouter()
 
-	client, err := p.grpcConn.Subscribe(ctx, &plumbing.SubscriptionRequest{
-		ShardID: firehoseSubscriptionId,
-	})
-	if err != nil {
-		writer.WriteHeader(http.StatusServiceUnavailable)
-		log.Printf("error occurred when subscribing to doppler: %s", err)
-		return
-	}
+	adminAccessMiddleware := NewAdminAccessMiddleware(adminAuthorizer)
+	logAccessMiddleware := NewLogAccessMiddleware(logAuthorizer)
 
-	p.serveWS(FIREHOSE_ID, firehoseSubscriptionId, writer, request, client)
-}
+	r.Handle("/set-cookie", NewSetCookieHandler(cookieDomain))
 
-// "^/apps/(.*)/(recentlogs|stream|containermetrics)$"
-func (p *DopplerProxy) serveAppLogs(requestPath, appID string, writer http.ResponseWriter, request *http.Request) {
-	authToken := getAuthToken(request)
-
-	status, _ := p.logAuthorize(authToken, appID)
-	if status != http.StatusOK {
-		switch status {
-		case http.StatusUnauthorized:
-			writer.WriteHeader(status)
-			writer.Header().Set("WWW-Authenticate", "Basic")
-		case http.StatusForbidden, http.StatusNotFound:
-			status = http.StatusNotFound
-		default:
-			status = http.StatusInternalServerError
-		}
+	containerMetricsHandler := NewContainerMetricsHandler(grpcConn, timeout)
+	r.Handle("/apps/{appID}/containermetrics", logAccessMiddleware.Wrap(
+		containerMetricsHandler,
+	))
 
-		writer.WriteHeader(status)
+	recentLogsHandler := NewRecentLogsHandler(grpcConn, timeout)
+	r.Handle("/apps/{appID}/recentlogs", logAccessMiddleware.Wrap(recentLogsHandler))
 
-		return
-	}
+	streamHandler := NewStreamHandler(grpcConn)
+	r.Handle("/apps/{appID}/stream", logAccessMiddleware.Wrap(streamHandler))
 
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
+	firehoseHandler := NewFirehoseHandler(grpcConn)
+	r.Handle("/firehose/{subID}", adminAccessMiddleware.Wrap(firehoseHandler))
```

Comment:
Do we expect this handler and the ones above to handle any HTTP verb? Or only `GET`?